src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -8413,617 +9036,195 @@ void SketchObject::rebuildExternalGeometry(bool defining, bool addIntersection)
-                                    else {
-                                        // P1 = P1, already defined
-                                        // P2 = P2, already defined
+                        if (fabs(dnormal.Angle(snormal) - M_PI_2) < Precision::Confusion()) {
+                            // The face is normal to the sketch plane
+                            // We don't want to keep the projection of all the edges of the face.
+                            // We need a single line that goes from min to max of all the projections.
+                            bool initialized = false;
+                            Base::Vector3d start, end;
+                            // Lambda to determine if a point should replace start or end
+                            auto updateExtremes = [&](const Base::Vector3d& point) {
+                                if ((point - start).Length() < (point - end).Length()) {
+                                    // `point` is closer to `start` than `end`, check if it's further out than `start`
+                                    if ((point - end).Length() > (end - start).Length()) {
+                                        start = point;
                                     }
                                 }
                                 else {
-                                    if (endAngle < 2 * M_PI) {
-                                        P1 = ProjPointOnPlane_XYZ(pntF, sketchPlane);
-                                        P2 = ProjPointOnPlane_XYZ(pntL, sketchPlane);
-                                    }
-                                    else if (endAngle < 4 * M_PI - startAngle) {
-                                        P1 = ProjPointOnPlane_XYZ(pntF, sketchPlane);
-                                        // P2 = P2, already defined
-                                    }
-                                    else if (endAngle < 3 * M_PI) {
-                                        // P1 = P1, already defined
-                                        P2 = ProjPointOnPlane_XYZ(pntL, sketchPlane);
-                                    }
-                                    else {
-                                        // P1 = P1, already defined
-                                        // P2 = P2, already defined
+                                    // `point` is closer to `end`, check if it's further out than `end`
+                                    if ((point - start).Length() > (end - start).Length()) {
+                                        end = point;
                                     }
                                 }
-                            }
+                            };
+                            for (auto& geo : geos) {
+                                auto* line = dynamic_cast<Part::GeomLineSegment*>(geo.get());
+                                if (!line) {
+                                    continue; // we should have only lines

```

Comment:
Comment suggests this should be `break` an perhaps an exception. I'll leave it to your judgment here.